lib/libvarnishapi/vsl_cursor.c

```diff
@@ -335,18 +335,20 @@ vslc_file_delete(const struct VSL_cursor *cursor)
 
 /* Read n bytes from fd into buf */
 static ssize_t
-vslc_file_readn(int fd, void *buf, ssize_t n)
+vslc_file_readn(int fd, void *bufa, ssize_t n)
 {
-	ssize_t t = 0;
+	char *buf = bufa;
 	ssize_t l;
 
-	while (t < n) {
-		l = read(fd, (char *)buf + t, n - t);
+	while (n > 0) {
+		l = read(fd, buf, n);
 		if (l <= 0)
 			return (l);
-		t += l;
+		assert(l <= n);
+		buf += l;
+		n -= l;
 	}
-	return (t);
+	return (n);
```

Comment:
Isn't this no longer returning the number of bytes read?

Reply:
I pushed too early and you were too quick (but I marked the PR as Draft). Yes, I noticed myself...